add.go

```diff
@@ -102,6 +103,30 @@ type AddAndCopyOptions struct {
 	Parents bool
 	// Timestamp is a timestamp to override on all content as it is being read.
 	Timestamp *time.Time
+	// Link, when set to true, creates an independent layer containing the copied content
+	// that sits on top of existing layers. This layer can be cached and reused
+	// separately, and is not affected by filesystem changes from previous intructions.
+	Link bool
+	// InheritLabels controls whether labels from the base image are preserved for --link operations.
+	// When set to OptionalBoolFalse, "|inheritLabels=false" is added to the
+	// createdBy history string to track this build-time decision.
+	InheritLabels types.OptionalBool
+	// UnsetAnnotations is a list of annotation keys to remove from the image for --link operations.
+	// Each annotation is recorded in the createdBy history as "|unsetAnnotation=key"
+	// to ensure proper cache invalidation when annotation removal differs between builds.
+	UnsetAnnotations []string
+	// InheritAnnotations controls whether annotations from the base image are preserved for --link operations.
+	// When set to OptionalBoolFalse, "|inheritAnnotations=false" is added to the
+	// createdBy history string to track this build-time decision.
+	InheritAnnotations types.OptionalBool
+	// Annotations is a list of key=value pairs to add as image annotations for --link operations.
+	// These are joined with commas and appended to the createdBy history string
+	// to ensure cache misses when annotation values differ between builds.
+	Annotations []string
+	// IsLastStep indicates whether this is the final instruction in the build for --link operations.
+	// When true, annotation and inheritance metadata is included in the createdBy
+	// history to ensure proper layer caching behavior.
+	IsLastStep bool
```

Comment:
Probably simpler to cook all of this down to a string that gets appended to the history entry when `Link` is set, and have the caller compute it in a helper function that's extracted from `getCreatedBy()`.  Then that knowledge won't have to be duplicated and kept in sync going forward.